byway/src/main/webapp/routepage.js

```diff
@@ -49,27 +71,105 @@ function calcRoute(directionsService, directionsRenderer, start, end) {
   });
 }
 
-function addToStops(stop){
-  console.log("add" + stop.placename + "to the ArrayList in the servlet");
+/** Clear the stops panel in the html */
+function clearStops(){
+  const stopList = document.getElementById('stop-list');
+  if(stopList != null){
+    stopList.innerText = "";
+  }
 }
 
-function getRecs() {
-  fetch('/api/recs')
+/** Get the new list of stops from datastore onload */
+function getStopsOnload(){
+  clearStops();
+  fetch('/api/stop')
   .then(response => response.json())
-  .then((recs) => {
-    const recList = document.getElementById('rec-list');
-    recList.innerText = ""; // clear old comments 
-    recs.forEach((stop) => {
-      let btn = document.createElement("button");
-      btn.innerText = stop.placename;
-      btn.setAttribute("class", "btn rec-btn");
-      btn.addEventListener("click", function() {
-        addToStops(stop);
-      });
-      recList.appendChild(btn);
-    });
+  .then((stopsResponse) => {
+    if(stopsResponse != null){
+      stops.push(...stopsResponse);
+    }
+    renderStopsList();
   });
 }
 
+/** Re-render stop list synchronously */
+function renderStopsList(){
+  clearStops();
+  const stopList = document.getElementById('stop-list');
+  stops.forEach((stop)=>{
+    stopList.appendChild(createStopButton(stop));
+  })
+}
+
+/** Render stop list  
+ *  @param {String} stop a String to add as a button in the schedule panel in the html
+ *  @return {button} stopBtn a button showing a selected stop
+ */
+function createStopButton(stop){
+  let stopBtn = document.createElement('button');
```

Comment:
Note that `const` in JS means "cannot assign a new value to the variable", it doesn't mean "can't modify the variable".

```js
const example = [];
example.push(3); // this is fine because it just modifies the value, doesn't assign to the variable.
example = [3]; // this doesn't work because it tries to assign a new value to the variable.
```

Aside from preventing re-assignment, `const` and `let` are identical, so you should always use `const` unless you specifically need to be able to assign a new value to the variable. Using the stricter variable type helps prevent bugs and makes the program easier to read and understand, because you know when you see `const` that the variable won't be reassigned anywhere, so it's a little easier to reason about. This is the essentially the same reason we always say to use `final` on fields of classes when working in Java.

Reply:
Oh I see, I thought it meant can't modify so I have been using let this whole time. Will change this and other occurences.